internal/service/config/nginx.go

```diff
@@ -191,3 +230,7 @@ func traverse(root *crossplane.Directive, callback crossplaneTraverseCallback, s
 	}
 	return nil
 }
+
+func getPermissions(fileMode os.FileMode) string {
+	return fmt.Sprintf("%#o", fileMode.Perm())
```

Comment:
not sure if this is the best place for this function

Reply:
Ya, I meant to ask about that. Donal had added it to `files.go` which I had deleted in my branch. I just put it there temporarily as it was only being used there. Forgot to ask where to put it, I can add files.go back in ?

Reply:
Added file_helpers.go which has GetPermissions()